carddav/client.go

```diff
@@ -429,3 +429,52 @@ func (c *Client) PutAddressObject(path string, card vcard.Card) (*AddressObject,
 	}
 	return ao, nil
 }
+
+// SyncCollection do a sync-collection operation on resource(path), it returns a SyncResponse
+func (c *Client) SyncCollection(path string, query *SyncQuery) (*SyncResponse, error) {
+	var limit *internal.Limit
+	if query.Limit > 0 {
+		limit = &internal.Limit{NResults: uint(query.Limit)}
+	}
+
+	propReq, err := encodeAddressPropReq(&query.DataRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	ms, err := c.ic.SyncCollection(path, query.SyncToken, internal.DepthOne, limit, propReq)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &SyncResponse{SyncToken: ms.SyncToken}
+	for _, resp := range ms.Responses {
+		p := resp.Hrefs[0].Path
+		if p == path || path == fmt.Sprintf("%s/", p) {
```

Comment:
I don't think the `/` workaround is necessary because address objects will be files, not directories.

Reply:
but the same directory present...
how do you suggest to solve it?

Reply:
What do you mean?

Reply:
without this practice, I still get the path concatenated with a trailing slash.

Reply:
Hmm. Alright, I don't see a better solution, let's keep the `/` workaround then.

Reply:
Opened https://github.com/emersion/go-webdav/issues/41